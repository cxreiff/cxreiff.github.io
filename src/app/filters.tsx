import { FC } from 'react'

export const Filters: FC = () => {
    return (
        <>
            <Turbulence />
        </>
    )
}

const Turbulence = () => (
    <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="0" height="0">
        <defs>
            <filter id="turbulence_0">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="0" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_1">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="1" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_2">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="2" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_3">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="3" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_4">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="4" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_5">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="5" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_6">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="6" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_7">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="7" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
        </defs> 
    </svg>
)
