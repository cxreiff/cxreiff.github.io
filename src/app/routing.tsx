import { FC, ReactNode, Suspense, useEffect, lazy } from "react";
import { Routes, Route, useLocation } from "react-router-dom";

import { Loader } from "src/common/loader/loader.component";
import { IFrame } from "src/common/iframe/iframe.component";
import {
  DOCUMENT_TITLE_ROOT,
  BREAKOUT_URL,
  PLATFORMER_URL,
  HEX_TILES_URL,
  PIXIJS_URL,
  BABYLONJS_URL,
  PLAIN_WEBGL_URL,
  BIGFOOT_MAP_URL,
} from "src/utilities/constants";

import Homepage from "src/features/homepage/homepage.component";
import Projects from "src/features/projects/projects.component";
import Photos from "src/features/photos/photos.component";
import Sketches from "src/features/sketches/sketches.component";
import ProjectsFigrid from "src/features/projects/projectsFigrid.component";

// const Posts = lazy(() => import("src/features/posts/posts.component"));
// const PostsContent = lazy(
//   () => import("src/features/posts/postsContent.component")
// );

const Shaders = lazy(() => import("src/features/shaders/shaders.component"));

const Pixijs = () => (
  <IFrame aspect={15.5 / 9} url={PIXIJS_URL} title={"pixijs"} canFullscreen />
);
const Babylonjs = () => (
  <IFrame
    aspect={3 / 2}
    url={BABYLONJS_URL}
    title={"babylonjs"}
    canFullscreen
  />
);
const Breakout = () => (
  <IFrame
    aspect={3 / 2}
    mobileAspect={2 / 3}
    url={BREAKOUT_URL}
    title={"breakout"}
    canFullscreen
  />
);
const Platformer = () => (
  <IFrame
    aspect={16 / 10}
    url={PLATFORMER_URL}
    title={"platformer"}
    canFullscreen
  />
);
const HexTiles = () => (
  <IFrame
    aspect={16 / 10}
    url={HEX_TILES_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const PlainWebGL = () => (
  <IFrame
    aspect={16 / 10}
    url={PLAIN_WEBGL_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const BigfootMap = () => (
  <IFrame aspect={16 / 10} url={BIGFOOT_MAP_URL} title={"bigfoot sightings"} />
);

export type RouteDefinition = {
  path: string;
  Component: FC;
  children?: RouteDefinition[];
};

export const ROUTES: RouteDefinition[] = [
  { path: "", Component: Homepage },
  {
    path: "projects",
    Component: Projects,
    children: [
      {
        path: "figrid",
        Component: ProjectsFigrid,
      },
      {
        path: "pixijs",
        Component: Pixijs,
        children: [{ path: ":id", Component: Pixijs }],
      },
      { path: "babylonjs", Component: Babylonjs },
      { path: "breakout", Component: Breakout },
      { path: "platformer", Component: Platformer },
      { path: "hex_tiles", Component: HexTiles },
      { path: "webgl", Component: PlainWebGL },
      { path: "bigfoot", Component: BigfootMap },
      {
        path: "shaders",
        Component: Shaders,
        children: [{ path: ":category", Component: Shaders }],
      },
      {
        path: "sketches",
        Component: Sketches,
        children: [{ path: ":id", Component: Sketches }],
      },
    ],
  },
  { path: "photos", Component: Photos },
  // {
  //   path: "posts",
  //   Component: Posts,
  //   children: [{ path: ":id", Component: PostsContent }],
  // },
];

export const Routing = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    const baseRoute = pathname.split("/")[1];
    document.title = baseRoute
      ? `${DOCUMENT_TITLE_ROOT} â€” ${baseRoute}`
      : DOCUMENT_TITLE_ROOT;
  }, [pathname]);

  const renderRoutes = (routes: RouteDefinition[], prefix = "") =>
    routes.reduce((renderedRoutes, { path, Component, children }) => {
      const fullPath = `${prefix}/${path}`;

      if (children) {
        renderedRoutes.push(...renderRoutes(children, fullPath));
      }

      return renderedRoutes.concat(
        <Route path={fullPath} element={<Component />} />
      );
    }, [] as ReactNode[]);

  return (
    <Suspense fallback={<Loader />}>
      <Routes>{renderRoutes(ROUTES)}</Routes>
    </Suspense>
  );
};
