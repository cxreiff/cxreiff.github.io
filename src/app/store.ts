import { createStore, applyMiddleware, compose } from 'redux'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import history from './history'
import homepage from '~/src/features/homepage/homepage.slice'

const router = connectRouter(history)

const middleware = [
    ...getDefaultMiddleware(),
    routerMiddleware(history),
];
 
if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middleware.push(logger);
}

const store = configureStore({
    reducer: {
        router,
        homepage,
    },
    middleware,
})

export type AppDispatch = typeof store.dispatch
export const useAppDispatch = () => useDispatch<AppDispatch>()

export type RootState = ReturnType<typeof store.getState>
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export default store

