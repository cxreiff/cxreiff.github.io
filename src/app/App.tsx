import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import 'normalize.css'

import store from './store'
import history from './history'

import Layout from '~/src/features/layout/layout.component'

const App: React.FC = () => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Layout/>
        </ConnectedRouter>
    </Provider>
)

export default App
