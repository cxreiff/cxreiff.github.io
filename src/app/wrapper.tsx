import React, { ComponentType, ReactElement } from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import store from './store'
import history from './history'

const wrapper = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapper
