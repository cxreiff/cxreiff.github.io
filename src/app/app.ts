import { FC } from 'react'
import 'normalize.css'

import wrapper from './wrapper'
import Layout from '~/src/features/layout/layout.component'

const App: FC = () => wrapper(Layout)

export default App
