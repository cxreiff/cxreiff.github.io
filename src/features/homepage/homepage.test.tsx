import { render } from '@testing-library/preact'

import { Wrapper } from '~/src/app/wrapper'
import Homepage from './homepage.component'

test('should render', () => {
    const { container, getByAltText } = render(<Homepage />, { wrapper: Wrapper })

    expect(getByAltText('image of site author Cooper Reiff')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
})
