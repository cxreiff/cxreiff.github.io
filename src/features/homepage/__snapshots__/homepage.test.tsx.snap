// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <section
    class="homepage"
  >
    <div
      class="frame image hidden"
    >
      <img
        alt="image of site author Jax Reiff"
        class="content"
        src="https://jaxreiff-static.sfo3.cdn.digitaloceanspaces.com/assets/aboutme.png"
      />
      <svg
        viewBox="0 0 1.5 1"
      />
    </div>
    <div
      class="aboutme"
    >
      <header>
        <h1>
          About Me
        </h1>
        <span
          class="theme_buttons"
        >
          <button
            aria-label="change theme to red color scheme"
            class="lightmode_button current_theme"
          />
          <button
            aria-label="change theme to green color scheme"
            class="greenish_button"
          />
          <button
            aria-label="change theme to blue color scheme"
            class="bluish_button"
          />
        </span>
      </header>
      <p>
        Hello! I am a software engineer specializing in front-end development. I have experience building clients with tools like TypeScript and React, and have expertise across the full stack as well.
      </p>
      <p>
        This is my personal site, where I've put photos, writing, and various projects I've been working on.
      </p>
      <p>
        The site itself also serves as an example (and test environment) of some of my preferred front-end technologies, libraries, and approaches. The source code for the site is publically available at its 
        <a
          href="https://github.com/jaxreiff/jaxreiff.github.io"
        >
          repository
        </a>
        , and links to resources and libraries that I have found useful building the site can be found in this 
        <a
          href="#/posts/portfolio-site-design"
        >
          post
        </a>
        .
      </p>
      <br />
      <p
        class="paragraph"
      >
        <span
          class="icon"
        >
          <svg
            class="lucide lucide-file-text"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.5 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7.5L14.5 2z"
            />
            <polyline
              points="14 2 14 8 20 8"
            />
            <line
              x1="16"
              x2="8"
              y1="13"
              y2="13"
            />
            <line
              x1="16"
              x2="8"
              y1="17"
              y2="17"
            />
            <line
              x1="10"
              x2="8"
              y1="9"
              y2="9"
            />
          </svg>
        </span>
        <span
          class="content"
        >
          My full-stack development resume can be found 
          <a
            href="https://jaxreiff.com/resume/resume.pdf#toolbar=0"
          >
            here
          </a>
          .
        </span>
      </p>
      <p
        class="paragraph"
      >
        <span
          class="icon"
        >
          <svg
            class="lucide lucide-github"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 22V18C15.14 16.75 14.78 15.49 14 14.5C17 14.5 20 12.5 20 9C20.08 7.75 19.73 6.52 19 5.5C19.28 4.35 19.28 3.15 19 2C19 2 18 2 16 3.5C13.36 3 10.64 3 8 3.5C6 2 5 2 5 2C4.7 3.15 4.7 4.35 5 5.5C4.27 6.52 3.92 7.75 4 9C4 12.5 7 14.5 10 14.5C9.61 14.99 9.32 15.55 9.15 16.15C8.98 16.75 8.93 17.38 9 18V22"
            />
            <path
              d="M9 18c-4.51 2-5-2-7-2"
            />
          </svg>
        </span>
        <span
          class="content"
        >
          My GitHub profile can be found 
          <a
            href="https://github.com/jaxreiff"
          >
            here
          </a>
          .
        </span>
      </p>
    </div>
  </section>
</div>
`;
